pdl-compiler/tests/generated/rust/packet_decl_grand_children_big_endian.rs

```diff
@@ -457,7 +457,7 @@ impl Packet for GrandChild {
                 maximum_size: 0xff,
             });
         }
-        buf.put_u8(2 + self.payload.len() as u8);
+        buf.put_u8((2 + self.payload.len()) as u8);
```

Comment:
This line looks like it could be a fix. However, on a second look,

```rust
(2 + self.payload.len()) as u8
```

and

```rust
2 + (self.payload.len() as u8)
```

should give the same result, right? Will the `2` literal not be type-inferred to be a `u8` here?

Is the problem with very large integers (larger than `i32`?) which can overflow in some way?

Reply:
Updated the added semantic test to properly check this output.

Reply:
Yes, now I see the problem clearly! Thanks 🙂